File: MotivApp/app/src/main/java/com/motiv/piotr/dao/PhotosListResponseDao.java
package com.motiv.piotr.dao;

import androidx.room.*;
import com.motiv.piotr.PhotosListResponseWithReferences;

@Dao
public interface PhotosListResponseDao {

    @Insert
    public long savePhotosListResponse(com.motiv.piotr.PhotosListResponse photosListResponse);

    @Delete
    public int deletePhotosListResponse(com.motiv.piotr.PhotosListResponse photosListResponse);

    @Query("SELECT * FROM photosListResponse")
    @Transaction
    public java.util.List<PhotosListResponseWithReferences> getPhotosListResponses();

    @Query("SELECT * FROM photosListResponse WHERE id = :id")
    @Transaction
    public PhotosListResponseWithReferences getPhotosListResponse(java.lang.String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    public void savePhotosListResponses(
            java.util.List<com.motiv.piotr.PhotosListResponse> photosListResponse);
}

File: MotivApp/app/src/main/java/com/motiv/piotr/dao/PhotosListResponseDao.kt
 
package com.motiv.piotr.dao
import androidx.room.*
import com.motiv.piotr.PhotosListResponse
import com.motiv.piotr.PhotosListResponseWithReferences
import kotlin.collections.List
@Dao
public interface PhotosListResponseDao {

    @Insert
    public fun savePhotosListResponse(photosListResponse: com.motiv.piotr.PhotosListResponse): Long@Delete
    public fun deletePhotosListResponse(photosListResponse: com.motiv.piotr.PhotosListResponse): Int@Query("SELECT * FROM photosListResponse")
    @Transaction
    public fun getPhotosListResponses(): List<PhotosListResponseWithReferences>@Query("SELECT * FROM photosListResponse WHERE id = :id")
    @Transaction
    public fun getPhotosListResponse(id: String): PhotosListResponseWithReferences@Insert(onConflict = OnConflictStrategy.REPLACE)
    public fun savePhotosListResponses(photosListResponse: List<com.motiv.piotr.PhotosListResponse>): Unit
}
